spring-cloud-gcp-bigquery/src/test/java/org/springframework/cloud/gcp/bigquery/integration/BigQueryTemplateIntegrationTests.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.bigquery.integration;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.ExecutionException;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobStatus;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableResult;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.gcp.bigquery.core.BigQueryTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.concurrent.ListenableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.cloud.gcp.bigquery.integration.BigQueryTestConfiguration.DATASET_NAME;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = BigQueryTestConfiguration.class)
+public class BigQueryTemplateIntegrationTests {
+
+	private static final String TABLE_NAME = "template_test_table";
+
+	@Autowired
+	BigQuery bigQuery;
+
+	@Autowired
+	BigQueryTemplate bigQueryTemplate;
+
+	@Before
+	public void setup() {
+		// Clear the previous dataset before beginning the test.
+		this.bigQuery.delete(TableId.of(DATASET_NAME, TABLE_NAME));
```

Comment:
I wonder how this test is going to work when I don't have access to GCP...

And why don't you clean DB after tests?..

Reply:
OK. Let them be integration tests, but how can you guarantee with their configuration that they are not going to be run for me locally unconditionally?